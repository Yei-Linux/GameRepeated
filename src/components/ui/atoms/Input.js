import classNames from 'classnames'
import React from 'react'

const Input = ({
  ariaLabel,
  border = true,
  onFocus,
  onBlur,
  variation = 'primary',
  name,
  defaultValue,
  onChange,
  placeholder,
  css,
  className,
  borderRadius = 1,
  padding = 1,
  onClick,
}) => {
  return (
    <input
      aria-label={ariaLabel}
      onClick={onClick}
      onBlur={onBlur}
      onFocus={onFocus}
      type="text"
      autoComplete="off"
      onChange={onChange}
      placeholder={placeholder}
      style={css}
      className={classNames(
        className,
        'input',
        `input--${variation}`,
        `input--border${borderRadius}`,
        `input--pd${padding}`,
        {
          'input--bordernone': !border,
        }
      )}
      name={name}
      value={defaultValue}
    />
  )
}

export default Input
