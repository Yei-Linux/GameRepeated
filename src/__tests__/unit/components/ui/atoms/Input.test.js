import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Input from '../../../../../components/ui/atoms/Input'

describe('The Input Component', () => {
  const defaultProps = {
    ariaLabel: 'buttonTest',
    border: true,
    variation: 'primary',
    name: 'buttonTest',
    defaultValue: '',
    placeholder: 'input',
    css: {},
    className: 'inputCustom',
    borderRadius: 1,
    padding: 1,
    onFocus: jest.fn(),
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onClick: jest.fn(),
  }

  const setupComponent = (props = defaultProps) => render(<Input {...props} />)

  it('Should reder input component', () => {
    const { asFragment } = setupComponent()

    expect(asFragment()).toMatchSnapshot()
  })
  it('Should reder input component', async () => {
    const mockOnChange = jest.fn(({ target: { value } }) => value)
    setupComponent({ ...defaultProps, onChange: mockOnChange })

    const input = screen.getByRole('textbox', { name: 'buttonTest' })
    userEvent.type(input, 'cesar')

    const valueCompared = mockOnChange.mock.results
      .map(({ value }) => value)
      .join('')

    expect(mockOnChange).toBeCalledTimes(5)
    expect(valueCompared).toStrictEqual('cesar')
  })
})
