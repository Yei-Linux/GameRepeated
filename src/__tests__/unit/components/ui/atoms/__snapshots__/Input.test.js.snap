// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Input Component Should reder input component 1`] = `
<DocumentFragment>
  <input
    aria-label="buttonTest"
    autocomplete="off"
    class="inputCustom input input--primary input--border1 input--pd1"
    name="buttonTest"
    placeholder="input"
    type="text"
    value=""
  />
</DocumentFragment>
`;
